### PR TITLE
Fix TidbDashboard baseImage parsing fault

### PR DESCRIPTION
**Pull request by linkinghack**
Fix wrong baseImage version adapt for image tag while the repo has specified a unique port with ':'.

### What problem does this PR solve?

Fix the bug that  baseImage auto version adapt can be executed unexpected while the repo has specified a unique port with ':'.

e.g. 
 If baseImage = example.registry.com:30000/foo/pingcap/tidb-dashboard:vx.y.z,  the final image will be "example.registry.com:vx.y.z".

### What is changed and how does it work?

Judge whether there were '\' after the last ':'.  If there is one, it is not a ':' for the image tag.

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Fix TidbDashboard baseImage parsing fault.
```

### CODE REVIEW

**Comment by linkinghack** [comment]
@handlerww @liubog2008  Please review this PR when you are free🫡

File: pkg/manager/tidbdashboard/manager.go [hunk 1]
```diff
@@ -347,7 +347,9 @@ func EnsureImage(td *v1alpha1.TidbDashboard) string {
 		version := td.Spec.Version
 		if version == nil || *version == "" {
 			image = baseImage
-		} else if l := strings.LastIndex(baseImage, ":"); l >= 0 {
+		} else if l, s := strings.LastIndex(baseImage, ":"), strings.LastIndex(baseImage, "/"); l >= 0 && l > s {
+			// i.e. example.registry.com:30000/foo/pincap/tidb-dashboard:vx.x.x
+			// If there is only one ':', it MUST NOT be the seprator of hostname and port.
```
Comment (csuzhangxc) [inline pkg/manager/tidbdashboard/manager.go:352]: I do not understand your comment here well. Do you mean for `example.registry.com:30000/foo/pincap/tidb-dashboard`, the whole part should be `baseImage` and no `verion` here, right? BTW, cloud you add some unit tests?

Comment (linkinghack) [reply pkg/manager/tidbdashboard/manager.go:352]: Yes, it is what I mean. 
I will add some tests.

Comment (linkinghack) [reply pkg/manager/tidbdashboard/manager.go]: @csuzhangxc  review again pls. 🥸

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**